### PR TITLE
remove unnecessary factory delegation for RESTClientGetter method

### PR DESCRIPTION
**Pull request by deads2k**
.Namespace is a method we don't need on the factory.  This makes it clear that the builders are being wired to the RESTClientGetter.

@kubernetes/sig-cli-maintainers 
/assign @juanvallejo

### CODE REVIEW

File: pkg/kubectl/cmd/logs_test.go [hunk 1]
```diff
@@ -123,6 +122,7 @@ func testPod() *api.Pod {
 func TestValidateLogFlags(t *testing.T) {
 	f := cmdtesting.NewTestFactory()
 	defer f.Cleanup()
+	f.WithNamespace("")
```
Comment (liggitt) [inline pkg/kubectl/cmd/logs_test.go]: this is odd

**Review by soltysh — APPROVED** [review]
lgtm, but it would be nice to change that empty `WithNamespace` with test as with all the rest tests

**Review by juanvallejo — APPROVED** [review]

File: pkg/kubectl/cmd/logs_test.go (hunk 1 above)
Comment (deads2k) [reply pkg/kubectl/cmd/logs_test.go]: > this is odd

ze germans

Comment (deads2k) [reply pkg/kubectl/cmd/logs_test.go]: > this is odd
> 
> ze germans

see, they struck!

**Comment by deads2k** [comment]
> lgtm, but it would be nice to change that empty WithNamespace with test as with all the rest tests

Something about that test wants it explicitly set.  I don't want to fight the test mocks at this point.

**Comment by soltysh** [comment]
> Something about that test wants it explicitly set. I don't want to fight the test mocks at this point.

sgtm
/lgtm

**Pull request closed without merging**